File: remote/src/main/java/com/kelsos/mbrc/rest/requests/RepeatRequest.java
package com.kelsos.mbrc.rest.requests;

import com.kelsos.mbrc.annotations.RepeatMode;

public class RepeatRequest {
  @RepeatMode private final String mode;

  public RepeatRequest(@RepeatMode String mode) {
    this.mode = mode;
  }

  @RepeatMode
  public String getMode() {
    return mode;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/dto/requests/RepeatRequest.kt
package com.kelsos.mbrc.dto.requests

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.kelsos.mbrc.annotations.Repeat
import com.kelsos.mbrc.annotations.Repeat.Mode

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("mode")
class RepeatRequest {

    @Mode
    @JsonProperty("mode")
    var mode: String = Repeat.NONE

}
